examples/pytorch/dis_kvstore/server.py

```diff
@@ -1,42 +1,55 @@
-# This is a simple MXNet server demo shows how to use DGL distributed kvstore.
-import dgl
+import os
 import argparse
+import time
+
+import dgl
+from dgl.contrib import KVServer
+
 import torch as th
 
-ndata_g2l = []
-edata_g2l = []
+g2l = []
+g2l.append(th.tensor([0,1,0,0,0,0,0,0]))
+g2l.append(th.tensor([0,0,0,1,0,0,0,0]))
+g2l.append(th.tensor([0,0,0,0,0,1,0,0]))
+g2l.append(th.tensor([0,0,0,0,0,0,0,1]))
+
+data = []
+data.append(th.tensor([[4.,4.,4.],[4.,4.,4.]]))
+data.append(th.tensor([[3.,3.,3.],[3.,3.,3.]]))
+data.append(th.tensor([[2.,2.,2.],[2.,2.,2.]]))
+data.append(th.tensor([[1.,1.,1.],[1.,1.,1.]]))
+
+class ArgParser(argparse.ArgumentParser):
+    def __init__(self):
+        super(ArgParser, self).__init__()
+
+        self.add_argument('--machine_id', type=int, default=0,
+                          help='Unique ID of each machine.')
+        self.add_argument('--server_id', type=int, default=0,
+                          help='Unique ID of each server.')
+        self.add_argument('--ip_config', type=str, default='ip_config.txt',
+                          help='IP configuration file of kvstore.')
+        self.add_argument('--backup_count', type=int, default=2,
+                          help='Count of backup server.')
 
-ndata_g2l.append({'ndata':th.tensor([0,1,0,0,0,0,0,0])})
-ndata_g2l.append({'ndata':th.tensor([0,0,0,1,0,0,0,0])})
-ndata_g2l.append({'ndata':th.tensor([0,0,0,0,0,1,0,0])})
-ndata_g2l.append({'ndata':th.tensor([0,0,0,0,0,0,0,1])})
+def start_server(args):
+    """Start kvstore service
+    """
+    server_namebook = dgl.contrib.read_ip_config(filename=args.ip_config)
 
-edata_g2l.append({'edata':th.tensor([0,1,0,0,0,0,0,0])})
-edata_g2l.append({'edata':th.tensor([0,0,0,1,0,0,0,0])})
-edata_g2l.append({'edata':th.tensor([0,0,0,0,0,1,0,0])})
-edata_g2l.append({'edata':th.tensor([0,0,0,0,0,0,0,1])})
+    my_server = KVServer(server_id=args.server_id, server_addr=server_namebook[args.server_id], num_client=8)
 
-DATA = []
-DATA.append(th.tensor([[4.,4.,4.,],[4.,4.,4.,]]))
-DATA.append(th.tensor([[3.,3.,3.,],[3.,3.,3.,]]))
-DATA.append(th.tensor([[2.,2.,2.,],[2.,2.,2.,]]))
-DATA.append(th.tensor([[1.,1.,1.,],[1.,1.,1.,]]))
+    if args.server_id % args.backup_count == 0:
+        my_server.set_global2local(name='entity_embed', global2local=g2l[args.machine_id])
+        my_server.init_data(name='entity_embed', data_tensor=data[args.machine_id])
+    else:
+        time.sleep(3)
+        my_server.set_global2local(name='entity_embed', global2local=None, data_shape=tuple((8,)))
+        my_server.init_data(name='entity_embed', data_tensor=None, data_shape=tuple((2,3)))
```

Comment:
I'm wondering if this is an easy API for people to use. In a real case, how does a user get the data shape? we cannot ask a user to hard code the shape.

Reply:
Yes, it could be a question. But the name-shared tensor API requires the user to provide the data shape. One solution is that the server node can communicate with each other and share the data_shape from the master server. But this can involve complicated logic on server nodes.

Reply:
I think the master server can write the data_shape information into a tmp file and the other servers can read from that.

Reply:
Can we just automatically generate a config file for all these kinds of info and sent to all servers along with the partition?

Reply:
another option is that the so-called "backup" server is created by the master server in the machine. In this way, the master server loads node/edge features in the file and the "backup" server gets the tensors automatically.